### PR TITLE
Add callback parameter to execute when signal is received

### PR DESCRIPTION
**Pull request by KmeCnin**
It is very useful to be able to call a custom function as soon as the signal starting the monitoring is received.  
It allows for instance to be able to print a message like  
`"Blackfire received a signal. Starting to monitor command."`

### CODE REVIEW

File: src/Blackfire/LoopClient.php [hunk 1]
```diff
@@ -40,17 +40,21 @@ public function __construct(Client $client, $maxIterations)
     }
 
     /**
-     * @param int $signal A signal that triggers profiling (like SIGUSR1)
+     * @param int      $signal   A signal that triggers profiling (like SIGUSR1)
+     * @param callable $callback A function to execute as soon as the signal is received
      */
-    public function setSignal($signal)
+    public function setSignal($signal, $callback = null)
```
Comment (Possum3D) [inline src/Blackfire/LoopClient.php]: callable $callback = null

File: src/Blackfire/LoopClient.php [hunk 2]
```diff
@@ -40,17 +40,21 @@ public function __construct(Client $client, $maxIterations)
...
     {
         if (!extension_loaded('pcntl')) {
             throw new RuntimeException('pcntl must be available to use signals.');
         }
 
         $enabled = &$this->enabled;
-        pcntl_signal($signal, function ($signo) use (&$enabled) {
+        pcntl_signal($signal, function ($signo) use (&$enabled, $callback) {
             $enabled = true;
+            if (is_callable($callback)) {
```
Comment (Possum3D) [inline src/Blackfire/LoopClient.php]: if (!$callback) {
return;
}

($callback)();

**Comment by KmeCnin** [comment]
:up:

**Comment by KmeCnin** [comment]
:up:

**Comment by romainneutron** [comment]
@KmeCnin Please stop spamming with comments you remove afterward

**Pull request closed without merging**